modules/chapter2/pages/git-auth.adoc

```diff
@@ -0,0 +1,226 @@
+. In the `Register new GitHub App` page, fill out the following details:
+
+* `GitHub App name`: *rhdh-app-<your_RHDP_GUID>* (A unique name)
+* `Homepage URL`: *https://rhdh-developer-hub-devhub.apps.cluster-<GUID>.dynamic.redhatworkshops.io* (Your RHDH Route URL)
+* `Callback URL`: *https://rhdh-developer-hub-devhub.apps.cluster-<GUID>.dynamic.redhatworkshops.io/api/auth/github/handler/frame*
+* `Webhook URL`: *https://rhdh-developer-hub-devhub.apps.cluster-<GUID>.dynamic.redhatworkshops.io*
+* `Webhook secret`: *5up3rS3cret* (Provide a strong secret)
+* Under `Permissions`, expand `Repository Permissions` and ensure that you provide at least `Read-only` access to the GitHub features you would like to integrate from RHDH. At a minimum, toggle the following permissions:
+** Administration
+** Commit statuses
+** Contents
+** Dependabot
+** Deployments
+** Pull Requests
+** Webhooks
+* Toggle other permissions as per your needs.
+* Under `Organization permissions`, ensure that you toggle `Read-only` permissions for `Members` so that RHDH can import team and user information from GitHub.
+* Select `Only on this account` for the `Where can this GitHub App be installed?` option.
+. Verify your form entries, and when you are satisfied click `Create GitHub App`.
+. You will be shown your `App ID` and a client ID. Make note of these values for the next step.
+. Click `Generate a new client secret`. Copy the generated secret for the next steps.
++
+image::2-gh-auth.png[title=Generate Client Secret,width=600]
+
+. Scroll down to the `Private Keys` section and click `Generate a private key`. Copy the generated private key certificate (it is in PEM format) to a safe location.
+. In the left sidebar menu of the generated app, select `Install App`, and then click `Install`.
+. Select `All Repositories` for `Repository access` and then click `Install`.
++
+image::8-gh-auth.png[title=Install GitHub App,width=400]
+
+### Create a Secret to Store GitHub App Credentials
+
+Rather than store the GitHub App credentials in plain text directly in the `app-config-rhdh` ConfigMap, you will create a secret to store the credentials. You will then refer to the credentials as environment variables in the ConfigMap.
+
+. Create a file named `env.sh` with the following environment variables and replace the values with the GitHub App details you created in the previous step.
++
+[subs=+quotes]
+----
+export OPENSHIFT_CLUSTER_INFO=$(oc cluster-info | head -n 1 | sed 's/^.*https...api//' | sed 's/.6443.*$//')
+export K8S_CLUSTER_API=$(oc cluster-info | head -n 1 |  sed 's/^.*https/https/')
+
+export GITHUB_HOST_DOMAIN=github.com
+export GITHUB_ORGANIZATION='Your Org Name'
+export GITHUB_ORG_URL=https://$GITHUB_HOST_DOMAIN/$GITHUB_ORGANIZATION
+
+export GITHUB_APP_ID='Your App ID'
+export GITHUB_APP_CLIENT_ID='Your client ID'
+export GITHUB_APP_CLIENT_SECRET='Your client secret'
+export GITHUB_APP_PRIVATE_KEY_FILE="$(< rhdh-app-<guid>.2024-01-17.private-key.pem)"
```

Comment:
Does it make sense to call out somewhere that you need to replace the <guid> in this file with your own?